helix-view/src/document.rs

```diff
@@ -38,6 +38,13 @@ pub enum Mode {
     Insert = 2,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Serialize)]
+pub enum FileStat {
```

Comment:
I think it would be more accurate to call this `WritePermissions` (and same for the field in `Document`): stat covers a lot of metadata and this is pretty specific.